### PR TITLE
signal if a blueprint wants a global filter

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@geirst @toregge please review
@havardpe FYI

next is adding the API for sending the global filter into a blueprint - i guess something like
setup_global_filter(BitVector::UP global_filter)
where the unique_ptr may be null to signal that there is no global filter (so we can assume all hits from ANN are actually returned as hits).

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍 I suggest we add a unit test in queryeval/intermediate_blueprints_test.cpp or queryeval/blueprint_test.cpp that verifies that the flag on an intermediate node is correctly inferred.

**Comment by geirst** [comment]
Perhaps we should call it set_global_filter()?, as the filter itself is already calculated at this point. With _setup_ it sounds that we should do something on the filter. I also think we should use a shared_ptr as the filter can be shared between multiple blueprints.

File: searchlib/src/vespa/searchlib/queryeval/blueprint.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ class Blueprint
         uint32_t          _cost_tier;
         uint32_t          _tree_size;
         bool              _allow_termwise_eval;
+        bool              _wants_global_filter;
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/blueprint.h:66]: please init in constructor

**Review by havardpe — APPROVED** [review]
I prefer 'want' over 'wants'

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**